kubernetes/kubernetes_service.go

```diff
@@ -135,9 +149,41 @@ func (in *IstioClient) GetServiceDetails(namespace string, serviceName string) (
 	}
 	serviceDetails.Autoscalers = autoscalersResponse.autoscalers
 
+	podsResponse := <-podsChan
+	if podsResponse.err != nil {
+		return nil, podsResponse.err
+	}
+	serviceDetails.Pods = podsResponse.pods
+
 	return &serviceDetails, nil
 }
 
+// GetServicePods returns the list of pods associated to a given service.
+// Pods belonging to a service are resolved by looking at the "app" label of the pods.
+// Optionally, a version can be specified to only return the pods belonging to the version (else,
+// set the version to an empty string).
+// It returns an error on any problem.
+func (in *IstioClient) GetServicePods(namespace, serviceName, serviceVersion string) (*v1.PodList, error) {
+	var labelSelectors []string
+
+	cfg := config.Get()
+	if serviceName != "" {
+		labelSelectors = append(labelSelectors, fmt.Sprintf("%v=%v", cfg.ServiceFilterLabelName, serviceName))
+	}
+	if serviceVersion != "" {
+		labelSelectors = append(labelSelectors, fmt.Sprintf("%v=%v", cfg.VersionFilterLabelName, serviceVersion))
+	}
+
+	podList, err := in.k8s.CoreV1().Pods(namespace).List(meta_v1.ListOptions{
+		LabelSelector: strings.Join(labelSelectors, ","),
```

Comment:
I am not fully convinced of this one.
I guess it should follow pattern defined here:
https://github.com/kiali/kiali/blob/d9e6f9529afda92f292afcc7c9ebe82889b7d4b8/kubernetes/kubernetes_service.go#L125

Fixed labels had problems and as I commented previously, at least we should use same selector defined in the service.

Reply:
I mean this in the context of the line:
https://github.com/kiali/kiali/pull/186/files#diff-3b12a0d5214c2182c3c6808c9b994122R121
If you are adding all the pods for a service, you should rely on the selector.

So, perhaps a suggestion is to define a common private implementation where you can pass the selector and create two methods, one for the ServiceDetails and other for specific version when you can build your own filter as you show here.

Probably we'll need both scenarios.

Reply:
@israel-hdez I think I know what @lucasponce is looking for, I will try to generate a supplement to consider for this PR.  I will try to have something for you to look at by the time you start today.

Reply:
@israel-hdez, @lucasponce I created https://github.com/israel-hdez/swscore/pull/1 as a PR to this PR to hopefully address Lucas's comments.  It does make sense to get pods using the Service's defined selector.  But we can also have an override available if we want to get pods for a specific version.  If it looks good maybe @israel-hdez can take this for his PR.